### DISCUSSION TITLE
Add a guide to help users to build a component library

### DISCUSSION THREAD

**Discussion by Barbapapazes** [vuejs/docs#3047] [Ideas]
Hello,

This is just an idea. What do you think of having a guide on how to create a Vue.js component library?

I know there is a demand for it, but it's not really well documented on the internet.

_And I would be happy to help to write it!_

related to https://github.com/vuejs/docs/issues/1748, https://github.com/vuejs/docs/issues/965, https://github.com/vuejs/docs/issues/1060

Edit:

I wrote a complete guide on my website: https://soubiran.dev/series/the-complete-guide-to-building-a-vue-js-component-library

**Comment by ZainW** [comment]
Happy to support you on this as well, 
i am a terrible writer but have created 3 component libraries from the beginning for companies I've worked for and have constantly pushed for better component building blocks

**Comment by Barbapapazes** [comment]
Not in the documentation but I wrote a complete guide: https://soubiran.dev/series/the-complete-guide-to-building-a-vue-js-component-library
